packages/format-library/src/link/index.js

```diff
@@ -43,6 +43,14 @@ function Edit( {
 	const [ openedBy, setOpenedBy ] = useState( null );
 
 	useLayoutEffect( () => {
+		// close the Link popover if there is no active selection
+		// after the link was added - this can happen if the user
+		// adds a link without any text selected
+		if ( addingLink && ! value.activeFormats ) {
```

Comment:
`!value.activeFormats` would mean there are no active formats of any kind (including bold, italitc...etc). Is this intended? If not then the `isActive` prop passed to this component might serve equally well:

https://github.com/WordPress/gutenberg/blob/345a938ef9e1b335a123901987e58a713f589a47/packages/block-editor/src/components/rich-text/format-edit.js#L40-L41

Reply:
Having _no activeFormats entry_ - as opposed to an empty `[]` list - in value is specific to when the selection is collapsed after the format is applied to the inserted link.

This is why it feels this fix is exploiting some other problem :)